src/webots/nodes/WbLidar.hpp

```diff
@@ -131,13 +131,14 @@ class WbLidar : public WbAbstractCamera {
   void initializeImageSharedMemory() override;
 
   int size() const override {
-    return sizeof(float) * width() * height() + sizeof(WbLidarPoint) * actualHorizontalResolution() * actualNumberOfLayers();
+    return sizeof(float) * actualHorizontalResolution() * height() +
```

Comment:
This structure should contain first the lidar image followed by the point cloud information.
Are you sure that this new size computation really matches the content of the shared memory?

Because the previous formula seems more correct.

Reply:
Isn't that the WbRenderingDevice::width() value should be updated and it is not?

Reply:
In fact the naming is a little bit misleading as `width()` is computed like this:
```
int WbLidar::width() const {
  if (isRotating())
    return ceil(actualHorizontalResolution() * (actualFieldOfView() / (2.0 * M_PI)));
  return actualHorizontalResolution();
}
```
So, in the case of a rotating lidar, it doesn't correspond anymore to the size of the lidar image, but to a smaller window.
Also, this line was recently changed in [#3032](https://github.com/cyberbotics/webots/commit/aaea21d899a03933eb1fc06d21754bbde805334e#diff-a8d078376f6e715ea6a984225234ddbbe25a58c20fdba74adad0ca623f3e4013L133) but it used to be `actualHorizontalResolution()` before.

Reply:
Ok. Thank you for the explanation.
I think that this class needs a deeper refactor because we should avoid misleading attribute names we should improve them.

Reply:
However, I just saw that in case of rotating lidars, the initial image is not stored in the shared memory but only put in `mTemporaryImage`, thus we might be able to define the size to `sizeof(float) * actualHorizontalResolution() * actualNumberOfLayers()`, plus the space for the point cloud.